station-world-events-v0/src/main/java/net/modificationstation/stationapi/api/celestial/CelestialEvent.java

```diff
@@ -0,0 +1,270 @@
+package net.modificationstation.stationapi.api.celestial;
+
+import lombok.Getter;
+import net.minecraft.world.World;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Contains the most important logic for automatically starting and stopping the event.
+ * It is recommended to add the event to CelestialTimeManger for automatic management.
+ * Inheritance is possible to add custom logic.
+ */
+public class CelestialEvent {
```

Comment:
Not the best name, considering `Event` suffix most likely indicates an UnsafeEvents event. Not sure what would be a good alternative though.

Reply:
I could replace the suffix with `Interval` which might actually be a better hint for what it is